### ISSUE TITLE
Tooltips bug

### ISSUE THREAD

**Issue by AndreiKingsley** [JetBrains/lets-plot-kotlin#176]
Tooltips for columns that mapped on x and y are displayed incorrectly
<img width="889" alt="image" src="https://user-images.githubusercontent.com/49092338/232450600-5dabbfb4-cc32-4577-bd5f-3e5b06d14cc4.png">

**Comment by alshan** [comment]
Taking in account that mapping `y = "b"` is ignored (unless stat=identity), this is likely an expected behavior.
@AndreiKingsley, what were you expecting in the tooltip?

**Comment by AndreiKingsley** [comment]
> Taking in account that mapping `y = "b"` is ignored (unless stat=identity), this is likely an expected behavior. @AndreiKingsley, what were you expecting in the tooltip?

Column names ("a", "b) on the left side of tooltips, just like for "c".

**Comment by alshan** [comment]
I see, at first I thought your complain is about tooltip showing "c: a1" instead "c: b2" :)

Macro '@' substituted with a default title for given aesthetic, which is `<empty>` for 'positional' aesthetics x/y.
@OLarionova-HORIS am I correct?

**Comment by AndreiKingsley** [comment]
But why is it empty?

**Comment by AndreiKingsley** [comment]
Is there any sense in this behavior?

**Comment by OLarionova-HORIS** [comment]
> Macro '@' substituted with a default title for given aesthetic, which is <empty> for 'positional' aesthetics x/y.
> @OLarionova-HORIS am I correct?

Yes, we don't show the default title for positional variables.
But it's possible to specify the title explicitly: `.line("title for a|@a").line("title for b|@b")`

**Comment by alshan** [comment]
May be we can invert this situation and for the setting `.line("@|@a")` render the tooltip line as 
> title  value 

unconditionally? 
And let the user to remove the title if necessary using `.line("|@a")` ?

**Comment by AndreiKingsley** [comment]
> May be we can invert this situation and for the setting `.line("@|@a")` render the tooltip line as
> 
> > title  value
> 
> unconditionally? And let the user to remove the title if necessary using `.line("|@a")` ?

Seems great!

**Comment by OLarionova-HORIS** [comment]
So the expected behaviour should be the following:
- `.line(@|^aesName).line(@|@varName)` => always `"title value"`
- `.line(|^aesName).line(|@varName)` => always `"value"`

But what about titles for positionals in the following situations:
- without specified tooltips
- tooltips are set via variable list: `layerTooltips( "a", "b" )` 

Now we use empty titles for them.

**Comment by alshan** [comment]
> without specified tooltips
tooltips are set via variable list: layerTooltips( "a", "b" )

I think, in the case of multiline tooltip it's preferable to show title regardless of the kind of the aesthetic.

**Comment by alshan** [comment]
> tooltips are set via variable list: layerTooltips( "a", "b" )

Another way to look at it: 
```
layerTooltips( "a", "b" )
```
is equivalent to 
```
layerTooltips()
  .line("@|@a")
  .line("@|@b")
```

**Comment by alshan** [comment]
Fixed in [LPK v4.4.0](https://github.com/JetBrains/lets-plot-kotlin/blob/master/CHANGELOG.md#440---2023-05-10)
